Use source heuristic in download_build

lslebodn opened the pull request:
We use source heuristic in download_build_data and save it info
playground/metadata/buildinfo.json.
    
Later, we got build_info one more time from koji in the function
download_build, but we forgot to use source heuristic there.
    
      Traceback (most recent call last):
        File "/src/scripts/download-unpack-build.py", line 53, in <module>
          utils.download_source(build_info, output_source_dir)
        File "/src/assayist/processor/utils.py", line 235, in download_source
          source_url = re.sub(r'^git\+http', r'http', build_info['source'])
        File "/usr/lib64/python3.6/re.py", line 191, in sub
          return _compile(pattern, flags).sub(repl, string, count)
      TypeError: expected string or bytes-like object
    
      Uncaught exception. Entering post mortem debugging
      Running 'cont' or 'step' will restart the program
      > /usr/lib64/python3.6/re.py(191)sub()
      -> return _compile(pattern, flags).sub(repl, string, count)
      (Pdb) p string
      None

lslebodn commented:
BTW other alternative would be to read file `metadata/buildinfo.json` or return `buildinfo` from `utils.download_build_data` and pass it into `utils.download_build`

mprpic commented:
> ...or return `buildinfo` from `utils.download_build_data` and pass it into `utils.download_build`

This seems like a better alternative. We already do this for `download_source` so we can just move up returning `build_info` one command earlier and reuse it for both `download_build` and `download_source`.

lslebodn committed:
Reuse koji build info from download_build_data

We use source heuristic in download_build_data and save it info
playground/metadata/buildinfo.json.

Later, we got build_info one more time from koji in the function
download_build, but we forgot to use source heuristic there.

The new version of function download_build_data returns modified
build_info using source heuristic and it is used in the function
download_source instead of just build_id.

  Traceback (most recent call last):
    File "/src/scripts/download-unpack-build.py", line 53, in <module>
      utils.download_source(build_info, output_source_dir)
    File "/src/assayist/processor/utils.py", line 235, in download_source
      source_url = re.sub(r'^git\+http', r'http', build_info['source'])
    File "/usr/lib64/python3.6/re.py", line 191, in sub
      return _compile(pattern, flags).sub(repl, string, count)
  TypeError: expected string or bytes-like object

  Uncaught exception. Entering post mortem debugging
  Running 'cont' or 'step' will restart the program
  > /usr/lib64/python3.6/re.py(191)sub()
  -> return _compile(pattern, flags).sub(repl, string, count)
  (Pdb) p string
  None

lslebodn commented:
flake8 passed; so it is ready for review

mprpic commented on assayist/processor/utils.py:
```diff
@@ -123,25 +125,25 @@ def download_build_data(build_identifier, output_dir='/metadata'):
 
-def download_build(build_identifier, output_dir):
+
+def download_build(build, output_dir):
```
Can we call this `build_info` to stay consistent with all other functions?

lslebodn replied:
Sure; I used `build` because it was previously used internally as a local variable.

mprpic reviewed (approved):
Looks good!

lslebodn committed:
Rename parameter in the function download_build

Other functions use build_info and it is better to be consistent.

lslebodn commented:
Re-based on top of current master

mprpic commented:
:+1:
